source/Coop/Mod/Persistence/RPC/EventBroadcastingQueue.cs

```diff
@@ -98,15 +98,17 @@ public void Update(TimeSpan frameTime)
         /// </summary>
         /// <param name="room">Room to send the event to.</param>
         /// <param name="rpc"></param>
-        public void Add(RailServerRoom room, EventMethodCall rpc)
+        public bool Add(RailServerRoom room, EventMethodCall rpc)
         {
             lock (m_Queue)
             {
                 Call call = new Call(room, rpc);
                 call.ObjectsToBeDistributed.RemoveAll(id => m_DistributedObjects.Contains(id));
 
+                bool isFull = false;
                 if (m_Queue.Count >= MaximumQueueSize)
                 {
+                    isFull = true;
                     Logger.Error("Event queue is full!");
```

Comment:
The error should only appear when the queue is full and an event is thus lost. In the best case scenario, this will be prevented by the pause. But i would assume that the pause will most likely trigger too late.

Reply:
Is it okay for some event to be lost or should I found a way to have the pause trigger before it's too late?